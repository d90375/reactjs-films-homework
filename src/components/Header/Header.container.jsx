import React, { useEffect } from "react";
import { useSelector } from "react-redux";
import Header from "./Header";
import { getDataByIdSelector } from "../../modules/headerData/headerSelectors";

const HeaderContainer = () => {
  const headData = useSelector(getDataByIdSelector);

  let { genres, runtime } = headData;
  if (genres) {
    genres = Object.values(genres)
      .reduce((acc, genre) => {
        return `${acc}${genre.name}, `;
      }, "")
      .slice(0, -2);
  }

  if (runtime) {
    const hours = runtime / 60;
    const calculatedHours = Math.floor(hours);
    const minutes = (hours - calculatedHours) * 60;
    const calculatedMinutes = Math.round(minutes);
    runtime = [calculatedHours, calculatedMinutes];
  }

  useEffect(() => {}, []);

  return (
    <>
      <Header headData={headData} genres={genres} runtime={runtime} />
    </>
  );
};

export default HeaderContainer;
