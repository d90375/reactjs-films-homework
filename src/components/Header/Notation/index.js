import Notation from "./Notation.container";

export default Notation;
