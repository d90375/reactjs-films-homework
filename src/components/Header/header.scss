.header {
  background-size: cover;
  display: flex;
  flex-direction: column;
  height: 700px;
  justify-content: space-between;
  width: 100%;

  @media (max-width: 1024px) {
    height: 580px;
  }

  @media (max-width: 767px) {
    height: 500px;
  }
}

.wrap {
  background-color: #f5f6f6;
  height: 700px;
  width: 100%;
}

.linkLogo {
  text-decoration: none;
}

.logo {
  color: #fff;
  font-family: "HelveticaNeue", serif;
  font-size: 2.5rem;
  letter-spacing: 1.37px;
  line-height: 64.29px;
  margin: 0;
  text-transform: uppercase;
}

.topWrap {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0 4rem 0 4rem;
  padding-top: 1rem;

  @media (max-width: 768px) {
    margin: 0 1rem 0 1rem;
  }

  @media (max-width: 500px) {
    flex-direction: column;
  }
}

.btmWrap {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0  0 2rem  4rem;

  @media (max-width: 767px) {
    flex-direction: column;
    margin: 0  1rem 2rem  1rem;
  }
}

.error {
  composes: error from "../../scss/core.scss";
}
