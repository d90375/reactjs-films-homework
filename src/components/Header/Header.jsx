import React from "react";
import PropTypes from "prop-types";
import Search from "./Search";
import Description from "./Description";
import Notation from "./Notation";
import { IMAGE_NOT_FOUND_URL } from "../../constants";

import "./Header.scss";

const Header = ({ headData, genres, runtime }) => {
  const { title, id } = headData;
  const voteAverage = headData.vote_average;
  const poster = headData.backdrop_path
    ? `url("https://image.tmdb.org/t/p/w1280${headData.backdrop_path}")`
    : IMAGE_NOT_FOUND_URL;
  const backGroundHeaderStyle = {
    backgroundRepeat: "no-repeat",
    backgroundPosition: "top",
    backgroundImage: `linear-gradient(0deg, rgba(0, 0, 0, 0.9) 2%, rgba(0, 0, 0, 0.2) 33%), ${poster}`,
    backgroundSize: "cover"
  };
  return (
    <header className="header" style={backGroundHeaderStyle}>
      <div className="header__topWrap">
        <h1 className="logo">FILMS</h1>
        <Search />
      </div>
      <div className="header__btmWrap">
        <Description title={title} genres={genres} runtime={runtime} voteAverage={voteAverage} />
        <Notation movieId={id} />
      </div>
    </header>
  );
};

export default Header;

Header.propTypes = {
  genres: PropTypes.string,
  headData: PropTypes.shape({
    adult: PropTypes.bool,
    backdrop_path: PropTypes.string,
    budget: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.object),
    homepage: PropTypes.string,
    id: PropTypes.number,
    imdb_id: PropTypes.string,
    original_language: PropTypes.string,
    original_title: PropTypes.string,
    overview: PropTypes.string,
    popularity: PropTypes.number,
    poster_path: PropTypes.string,
    production_companies: PropTypes.arrayOf(PropTypes.object),
    production_countries: PropTypes.arrayOf(PropTypes.object),
    release_date: PropTypes.string,
    revenue: PropTypes.number,
    spoken_languages: PropTypes.arrayOf(PropTypes.object),
    status: PropTypes.string,
    tagline: PropTypes.string,
    title: PropTypes.string,
    video: PropTypes.bool,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number
  }).isRequired,
  runtime: PropTypes.arrayOf(PropTypes.number)
};

Header.defaultProps = {
  genres: "",
  runtime: [0, 0]
};
