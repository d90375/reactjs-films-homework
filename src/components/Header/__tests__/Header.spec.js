import React from "react";
import Header from "../Header";

const setUp = (props) => {
  shallow(<Header {...props} />);
};

describe("Header component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render Header component without props ", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render Header component with props ", () => {
    component = setUp();
    expect(component).toMatchSnapshot();
  });
});
