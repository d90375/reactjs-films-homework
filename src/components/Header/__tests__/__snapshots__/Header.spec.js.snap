// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component with props  1`] = `undefined`;

exports[`Header component should render Header component without props  1`] = `undefined`;
